src/include/daos_pool.h

```diff
@@ -256,10 +256,13 @@ daos_pool_global2local(d_iov_t glob, daos_handle_t *poh);
 
 /**
  * Query pool information. User should provide at least one of \a info and
- * \a tgts as output buffer.
+ * \a ranks as output buffer.
  *
  * \param[in]	poh	Pool connection handle.
- * \param[out]	tgts	Optional, returned storage targets in this pool.
+ * \param[out]	ranks	Optional, returned storage ranks in this pool.
+ *			If any pool targets are disabled, \a ranks is the list of affected engines.
+ *			If no pool targets are disabled, \a ranks is the list of all engines.
```

Comment:
[Question] Does "disabled" mean "excluded"? Suppose rank 3 has 4 targets, one of which is excluded, and all other ranks as well as their targets are not excluded. Then the ranks parameter will return {3}?

Suppose I don't know if there's any excluded target or rank, and I get {0, 1} in the ranks parameter. How do I know if rank 0 and rank 1 are "all engines" or they are the engines with excluded targets?

Reply:
In the existing client code that searches for failed (disabled) targets in the pool map, disabled means targets reported to be in the DOWN state or DOWNOUT state. info.pi_ndisabled is set based on this count. Targets that are excluded do in practice cause that pi_ndisabled quantity to be incremented accordingly. And subsequently if reintegrated, pi_ndisabled decremented. So excluded targets I guess are considered disabled, although not sure if it is a "==" relationship between excluded & disabled. It may be, but there are some important comments in the code about draining/drained targets (not considered disabled while draining).

-- A statement about the intent of the patch for the API:
In the example of rank 3 with a single target excluded, yes the ranks parameter will return a list of a single rank (rank 3). This scenario will correspond to info.pi_ndisabled > 0 (that may have been caused, for example, by manual or auto-exclusion of the targets). In the example of ranks containing 0,1 then if info.pi_ndisabled == 0 ranks 0 and 1 can be interpreted as all engines (none of which, as it happens, have targets in a DOWN or DOWN_OUT state). Otherwise with pi_ndisabled > 0 then the ranks are interpreted as those having some/all of their targets disabled.

-- about the patch
I will look again at how the patch decides to populate the ranks parameter align it more precisely with disabled vs. not disabled targets. i.e., the pool_map_get_ranks() code could instead be written to either look for "disabled" targets (DOWN or DOWNOUT) -- or the converse (that I assume would include UP, UPIN, DRAIN). And clean up terminology as needed (up/down, excluded, disabled, ...).

Reply:
> In the example of ranks containing 0,1 then if info.pi_ndisabled == 0 ranks 0 and 1 can be interpreted as all engines (none of which, as it happens, have targets in a DOWN or DOWN_OUT state). Otherwise with pi_ndisabled > 0 then the ranks are interpreted as those having some/all of their targets disabled.

I see. I forgot about pi_ndisabled. Thanks for the explanation.